lib/services/amazonka-dynamodb-streams/src/Amazonka/DynamoDBStreams/Internal.hs

```diff
@@ -0,0 +1,169 @@
+-- Portability : non-portable (GHC extensions)
+module Amazonka.DynamoDBStreams.Internal
+  ( -- * Attribute Values
+    AttributeValue (..),
+
+    -- ** Attribute Value Prisms
+    _L,
+    _NS,
+    _M,
+    _NULL,
+    _N,
+    _BS,
+    _B,
+    _SS,
+    _S,
+    _BOOL,
+  )
+where
+
+import Amazonka.Core
+import Amazonka.Prelude
+import Control.Lens.TH (makePrisms)
+import Data.Aeson (pairs)
+import Data.Hashable
+import Data.Map (Map)
+import Data.Vector (Vector)
+
+#if MIN_VERSION_aeson(2,0,0)
+import qualified  Data.Aeson.KeyMap as KeyMap
+#else
+import qualified  Data.HashMap.Strict as KeyMap
+#endif
+
+-- | Represents the data for an attribute.
+--
+-- DynamoDB sends and receives JSON objects which contain a single
+-- item whose key is a data type and the value is the data itself. We
+-- provide an actual sum type to interact with these.
+--
+-- For more information, see
+-- <https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes Data Types>
+-- in the /Amazon DynamoDB Developer Guide/.
+data AttributeValue
+  = -- | An attribute of type List. For example:
+    --
+    -- @\"L\": [ {\"S\": \"Cookies\"} , {\"S\": \"Coffee\"}, {\"N\", \"3.14159\"}]@
+    L (Vector AttributeValue)
+  | -- | An attribute of type Number Set. For example:
+    --
+    -- @\"NS\": [\"42.2\", \"-19\", \"7.5\", \"3.14\"]@
+    --
+    -- Numbers are sent across the network to DynamoDB as strings, to maximize
+    -- compatibility across languages and libraries. However, DynamoDB treats
+    -- them as number type attributes for mathematical operations.
+    NS (Vector Text)
+  | -- | An attribute of type Map. For example:
+    --
+    -- @\"M\": {\"Name\": {\"S\": \"Joe\"}, \"Age\": {\"N\": \"35\"}}@
+    M (Map Text AttributeValue)
+  | -- | An attribute of type Null. For example:
+    --
+    -- @\"NULL\": true@
+    NULL
+  | -- | An attribute of type Number. For example:
+    --
+    -- @\"N\": \"123.45\"@
+    --
+    -- Numbers are sent across the network to DynamoDB as strings, to maximize
+    -- compatibility across languages and libraries. However, DynamoDB treats
+    -- them as number type attributes for mathematical operations.
+    N Text
+  | -- | An attribute of type Binary Set. For example:
+    --
+    -- @\"BS\": [\"U3Vubnk=\", \"UmFpbnk=\", \"U25vd3k=\"]@
+    BS (Vector Base64)
+  | -- | An attribute of type Binary. For example:
+    --
+    -- @\"B\": \"dGhpcyB0ZXh0IGlzIGJhc2U2NC1lbmNvZGVk\"@
+    B Base64
+  | -- | An attribute of type String Set. For example:
+    --
+    -- @\"SS\": [\"Giraffe\", \"Hippo\" ,\"Zebra\"]@
+    SS (Vector Text)
+  | -- | An attribute of type String. For example:
+    --
+    -- @\"S\": \"Hello\"@
+    S Text
+  | -- | An attribute of type Boolean. For example:
+    --
+    -- @\"BOOL\": true@
+    BOOL Bool
+  deriving stock (Eq, Read, Show, Generic)
+  deriving anyclass (NFData)
+
+$(makePrisms ''AttributeValue)
```

Comment:
Perhaps we should rely on `generic-lens`/`generic-optics`, as we do for everything else?

Reply:
Yes, I think we should do that.